src/wallet/scriptpubkeyman.cpp

```diff
@@ -202,11 +202,11 @@ isminetype LegacyScriptPubKeyMan::IsMine(const CScript& script) const
     assert(false);
 }
 
-bool CWallet::Unlock(const CKeyingMaterial& vMasterKeyIn, bool accept_no_keys)
+bool LegacyScriptPubKeyMan::CheckDecryptionKey(const CKeyingMaterial& master_key, bool accept_no_keys)
 {
     {
         LOCK(cs_KeyStore);
-        if (!SetCrypted())
+        if (!mapKeys.empty())
```

Comment:
In commit "Refactor: Move encryption code between KeyMan and Wallet" (d96925bde5c103693e44787503dc3e45d1248da8)

Would be great if this check and the same check in `LegacyScriptPubKeyMan::AddCryptedKeyInner` could be made into asserts or `CHECK_NONFATAL`'s, because it seems like something would have to be very wrong for these to trigger. Or it would be helpful to have a comment saying when this is expected, if it is ever expected.

Reply:
Changed to asserts.